Enabling .NET 9 tests for SF SDK packages

plave0 opened the pull request:
This PR enables tests for .NET 9 builds. 

Additionally, we are moving away from the `buils.ps1` script, which uses `msbuild` under the hood. Instead, we are moving to the `dotnet` tool, in the aim of streamlining the build process and infrastructure. 

Internal ADO work item: https://dev.azure.com/msazure/One/_workitems/edit/29603235

plave0 committed:
Enabling testing for net9.0

plave0 committed:
Updating build pipeline

Updating build pipeline to be compatible with .NET 9.

plave0 committed:
Reverting back to single target tests

plave0 commented on .github/workflows/build.yml:
```diff
@@ -21,9 +21,13 @@ jobs:
       - name: Build Everything
         shell: powershell
-        run: ./build.ps1
+        run: dotnet build code.sln
```
I have tested building the whole solution using `dotnet`, and it works. The same change will be implemented in `service-fabric-services-and-actors-dotnet`.

olegsych replied:
Let's do `dotnet build buildAll.proj` and get rid of the build.ps1 as I've explained in the services and actors PR.

plave0 committed:
Fixing build pipeline

We are still unable to use code.sln for building the repo, since it
doesn't include SF.ActorsServices.Internal. We will insted build
buildAll.proj for now.

plave0 committed:
Merge branch 'develop' into user/pavleiri/enableDotNet9Tests
